### PR TITLE
Make request object optional for login methods

### PR DESCRIPTION
**Pull request by tnorling**
Addresses #2054

### CODE REVIEW

File: lib/msal-browser/src/utils/BrowserConstants.ts [hunk 1]
```diff
@@ -50,3 +50,7 @@ export enum TemporaryCacheKeys {
     REQUEST_PARAMS = "request.params",
     SCOPES = "scopes"
 }
+
+export const DEFAULT_REQUEST = {
```
Comment (pkanher617) [inline lib/msal-browser/src/utils/BrowserConstants.ts:54]: We should add a typing to this request, can be `AuthorizationUrlRequest`.

File: lib/msal-browser/src/utils/BrowserConstants.ts [hunk 2]
```diff
@@ -50,3 +50,7 @@ export enum TemporaryCacheKeys {
...
+    scopes: ["openid", "profile"]
```
Comment (pkanher617) [inline lib/msal-browser/src/utils/BrowserConstants.ts:55]: Use the constant vars instead of strings here.

Comment (pkanher617) [reply lib/msal-browser/src/utils/BrowserConstants.ts:55]: `Constants.OPENID_SCOPE, Constants.PROFILE_SCOPE`

Comment (tnorling) [reply lib/msal-browser/src/utils/BrowserConstants.ts]: Yep good call! Done.

File: lib/msal-browser/src/utils/BrowserConstants.ts [hunk 3]
```diff
@@ -50,3 +52,7 @@ export enum TemporaryCacheKeys {
     REQUEST_PARAMS = "request.params",
     SCOPES = "scopes"
 }
+
+export const DEFAULT_REQUEST: AuthorizationUrlRequest = {
+    scopes: [Constants.OPENID_SCOPE, Constants.PROFILE_SCOPE]
```
Comment (jasonnutter) [inline lib/msal-browser/src/utils/BrowserConstants.ts:57]: If you pass empty array will these scopes get added automatically?

Comment (jasonnutter) [reply lib/msal-browser/src/utils/BrowserConstants.ts:57]: We could have the logic downstream also use this same constant.

File: lib/msal-browser/src/app/PublicClientApplication.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ import { RedirectHandler } from "../interaction_handler/RedirectHandler";
 import { PopupHandler } from "../interaction_handler/PopupHandler";
 import { SilentHandler } from "../interaction_handler/SilentHandler";
 import { BrowserAuthError } from "../error/BrowserAuthError";
-import { BrowserConstants, TemporaryCacheKeys } from "../utils/BrowserConstants";
+import { BrowserConstants, TemporaryCacheKeys, DEFAULT_REQUEST } from "../utils/BrowserConstants";
 import { BrowserUtils } from "../utils/BrowserUtils";
 import { version } from "../../package.json";
 import { IPublicClientApplication } from "./IPublicClientApplication";
```
Comment (jasonnutter) [inline lib/msal-browser/src/app/PublicClientApplication.ts:43]: Can you make sure `IPublicClientApplication` is also updated to show these are optional?

File: lib/msal-browser/src/utils/BrowserConstants.ts (hunk 3 above)
Comment (sameerag) [reply lib/msal-browser/src/utils/BrowserConstants.ts:57]: I had the exact q. What is the conclusion?

Comment (sameerag) [reply lib/msal-browser/src/utils/BrowserConstants.ts:57]: Feel free to resolve if addressed.

Comment (tnorling) [reply lib/msal-browser/src/utils/BrowserConstants.ts:57]: @sameerag Yes they do get added downstream, PR was updated to have the downstream addition utilize this constant as well

**Pull request merged**